### PR TITLE
Add some MapConfigDsl test

### PR DESCRIPTION
**Pull request by amrsamii**

### CODE REVIEW

**Comment by amrsamii** [comment]
I figured out that `JUnit5` needs `maven-surefire-plugin` so that maven can run the tests. I want to make this commit but first build fails problem should be fixed. Maybe the `SONAR_TOKEN` that you have added has a problem?

**Comment by nfrankel** [comment]
In the log file:

> java.lang.IllegalStateException: You're not authorized to run analysis. Please contact the project administrator.

This is unrelated to Maven, but is related to SonarCloud.

File: pom.xml [hunk 1]
```diff
@@ -9,6 +9,8 @@
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <kotlin.version>1.3.41</kotlin.version>
+        <junit.jupiter.version>5.6.1</junit.jupiter.version>
```
Comment (nfrankel) [inline pom.xml:12]: Why create a property if the version is used only once?

**Review by nfrankel — CHANGES_REQUESTED** [review]

**Comment by amrsamii** [comment]
> In the log file:
> 
> > java.lang.IllegalStateException: You're not authorized to run analysis. Please contact the project administrator.
> 
> This is unrelated to Maven, but is related to SonarCloud.

yes I know, but there is another issue that Maven did not run the test, in the logs:
> T E S T S
> -------------------------------------------------------
> Running org.hazelcast.configdsl.MapConfigDslTest
> Tests run: 0, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.072 sec
> Results :
> Tests run: 0, Failures: 0, Errors: 0, Skipped: 0

I already fixed this issue but I didn't commit changes yet. I want to fix the issue of `SonarCloud` first. Maybe the problem in the token? I gave you permission to execute analysis

File: pom.xml (hunk 1 above)
Comment (amrsamii) [reply pom.xml:12]: okay I will remove these properties

**Comment by nfrankel** [comment]
I unfortunately cannot help you with that but with generic statements: read the log, analyze the cause, fix it. Sorry

**Comment by amrsamii** [comment]
Okay I will trace the problem.

**Comment by amrsamii** [comment]
Unfortunately, making SonarCloud analyze external PR (PR to a repo that is not the same as the one that is not configured with SonarCloud) is not supported. here is the issue of the feature [MMF-1371](https://jira.sonarsource.com/browse/MMF-1371). they are working on it. so this means that every time I make a PR to your repo, build will fails because SonarCloud analysis is not triggered. hence, PR analysis works only if I make a PR from a repo to the same repo, not from a forked repo to the original repo. 

Solutions:
- I can make two workflows: one that runs Maven and SonarCloud analysis and is triggered when I push to any branch, this way, my feature/fix branch on forked repo is analysed using SonarCloud and tests also run. if things are successful, I make a PR to your repo, then the second workflow is triggered which only runs Maven, after that, you review the PR and you can check SonarCloud analysis of the branch. if it is OK, then you merge the PR, and the first workflow will be triggered any SonarCloud will analyze the master branch.

- you can give me access to your repo so that I can make a branch on it and make a PR directly to the master and SonarCloud PR analysis will be triggered, but the SonarCloud project must be associated with your repo, not with my forked repo.

**Comment by nfrankel** [comment]
Thanks for your analysis

**Comment by amrsamii** [comment]
No problem. as you chose the second solution, you need to configure the repo with SonarCloud so that when I make a PR, SonarCloud analysis to the PR will run.

**Comment by nfrankel** [comment]
I tried successfully with an empty commit

**Comment by amrsamii** [comment]
analyzing master branch will work, but PR on this repo is considered to be external to the SonarCloud project since it is configured with the forked repo, not this repo. you need to configure this repo with a new SonarCloud project and I will migrate SonarCloud analysis

**Comment by nfrankel** [comment]
I've made you a collaborator of _my_ project. Hence, you could add it as an additional remote and do the PR from a branch from this repo to master. This would be the easiest I believe

**Comment by amrsamii** [comment]
yes I know, but this PR will not be analyzed from the current SonarCloud project because it is configured with the forked repo not this repo

**Comment by nfrankel** [comment]
Let's forget this PR. Just add the branch to the main repo and do a PR from there

**Comment by amrsamii** [comment]
Okay I will close this PR

**Pull request closed without merging**